File: exercises/week05/exercise-dp-two/src/abstractfactory/NYCParserFactory.java
package abstractfactory;

public class NYCParserFactory implements AbstractParserFactory {

    @Override
    public XMLParser getParserInstance(String parserType) {

        switch (parserType) {
            case "NYCERROR":
                return new NYCErrorXMLParser();
            case "NYCFEEDBACK":
                return new NYCFeedbackXMLParser();
            case "NYCORDER":
                return new NYCOrderXMLParser();
            case "NYCRESPONSE":
                return new NYCResponseXMLParser();
        }

        return null;
    }

}

File: exercises/week05/exercise-dp-two-scala/src/abstractfactory/NYCParserFactory.scala
package abstractfactory

class NYCParserFactory extends AbstractParserFactory {
  def getParserInstance(parserType: String): XMLParser = {
    parserType match {
      case "NYCERROR" =>
        return new NYCErrorXMLParser
      case "NYCFEEDBACK" =>
        return new NYCFeedbackXMLParser
      case "NYCORDER" =>
        return new NYCOrderXMLParser
      case "NYCRESPONSE" =>
        return new NYCResponseXMLParser
    }
  }
}